### PR TITLE
SSCS-10167: adding new tab to capture case outcome

### PR DESCRIPTION
**Pull request by amittal89**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/SSCS-10167

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: benefit/data/sheets/AuthorisationCaseEvent/AuthorisationCaseEvent-sandl-nonprod.json [hunk 1]
```diff
@@ -2,5 +2,6 @@
     {"LiveFrom": "10/06/2022", "CaseTypeID": "Benefit", "CaseEventID": "listingError", "UserRole": "caseworker-sscs-systemupdate", "CRUD": "CRUD"},
     {"LiveFrom": "10/06/2022", "CaseTypeID": "Benefit", "CaseEventID": "listingError", "UserRole": "caseworker-sscs-superuser", "CRUD": "CRUD"},
     {"LiveFrom": "01/06/2022", "CaseTypeID": "Benefit", "CaseEventID": "updateListingRequirements", "UserRole": "caseworker-sscs-clerk", "CRUD": "CRUD"},
-    {"LiveFrom": "01/06/2022", "CaseTypeID": "Benefit", "CaseEventID": "updateListingRequirements", "UserRole": "caseworker-sscs-superuser", "CRUD": "CRUD"}
+    {"LiveFrom": "01/06/2022", "CaseTypeID": "Benefit", "CaseEventID": "updateListingRequirements", "UserRole": "caseworker-sscs-superuser", "CRUD": "CRUD"},
+    {"LiveFrom": "04/07/2022", "CaseTypeID": "Benefit", "CaseEventID": "addCaseOutcome", "UserRole": "caseworker-sscs-superuser", "CRUD": "CRUD"}
```
Comment (dattipoeHMCTS) [inline benefit/data/sheets/AuthorisationCaseEvent/AuthorisationCaseEvent-sandl-nonprod.json:6]: Should we have more roles for all these? Currently on a super user would be able to perform the event and use the fields

File: benefit/data/sheets/CaseType/CaseType.json [hunk 2]
```diff
@@ -1 +1,2 @@
 [{"LiveFrom": "01/01/2018", "ID": "Benefit", "Name": "SSCS Case v5.14.27_${CCD_DEF_E}", "Description": "SSCS Benefit Cases", "JurisdictionID": "SSCS", "PrintableDocumentsUrl": "http://ccd-data-store-api-prod.service.core-compute-prod.internal/callback/jurisdictions/SSCS/case-types/Benefit/documents", "RetriesTimeoutURLPrintEvent": "5,10,15", "SecurityClassification": "PUBLIC"}]
```
Comment (dattipoeHMCTS) [inline benefit/data/sheets/CaseType/CaseType.json:1]: Don't forget to bump the version number. Same in VERSION.yaml

File: benefit/data/sheets/CaseTypeTab/CaseTabType-sandl-nonprod.json [hunk 3]
```diff
@@ -7,5 +7,7 @@
...
     {"LiveFrom": "", "CaseTypeID": "Benefit", "TabID": "listing_requirements_judge", "TabLabel": "Listing Requirements", "TabDisplayOrder": 24, "CaseFieldID": "overrideDuration", "TabFieldDisplayOrder": 9},
     {"LiveFrom": "", "CaseTypeID": "Benefit", "TabID": "listing_requirements_clerk", "TabLabel": "Listing Requirements", "TabDisplayOrder": 24, "CaseFieldID": "overrideDuration", "TabFieldDisplayOrder": 7},
-    {"LiveFrom": "", "CaseTypeID": "Benefit", "TabID": "listing_requirements_registrar", "TabLabel": "Listing Requirements", "TabDisplayOrder": 24, "CaseFieldID": "overrideDuration", "TabFieldDisplayOrder": 7}
+    {"LiveFrom": "", "CaseTypeID": "Benefit", "TabID": "listing_requirements_registrar", "TabLabel": "Listing Requirements", "TabDisplayOrder": 24, "CaseFieldID": "overrideDuration", "TabFieldDisplayOrder": 7},
+    {"LiveFrom": "", "CaseTypeID": "Benefit","TabID": "outcome_superuser", "TabLabel": "Outcome", "TabDisplayOrder": 30, "CaseFieldID": "caseOutcome", "TabFieldDisplayOrder": 1},
```
Comment (dattipoeHMCTS) [inline benefit/data/sheets/CaseTypeTab/CaseTabType-sandl-nonprod.json:11]: Same question as above

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]

File: benefit/data/sheets/AuthorisationCaseEvent/AuthorisationCaseEvent-sandl-nonprod.json (hunk 1 above)
Comment (amittal89) [reply benefit/data/sheets/AuthorisationCaseEvent/AuthorisationCaseEvent-sandl-nonprod.json:6]: Authorisation for caseworker & superuser added.

File: benefit/data/sheets/CaseType/CaseType.json (hunk 2 above)
Comment (amittal89) [reply benefit/data/sheets/CaseType/CaseType.json:1]: Version was bumped in both files, they are in sync

File: benefit/data/sheets/AuthorisationCaseEvent/AuthorisationCaseEvent-sandl-nonprod.json (hunk 1 above)
Comment (dattipoeHMCTS) [reply benefit/data/sheets/AuthorisationCaseEvent/AuthorisationCaseEvent-sandl-nonprod.json:6]: You'll need to add the same for the event

File: benefit/data/sheets/CaseEvent/CaseEvent-sandl-nonprod.json [hunk 4]
```diff
@@ -9,5 +9,16 @@
...
     "SecurityClassification": "Public"
+  },
+  {
+    "LiveFrom": "04/07/2022",
+    "CaseTypeID": "Benefit",
+    "ID": "addCaseOutcome",
+    "Name": "Add case outcome",
+    "Description": "Add a case outcome",
+    "DisplayOrder": 14,
+    "PreConditionState(s)": "*",
```
Comment (dattipoeHMCTS) [inline benefit/data/sheets/CaseEvent/CaseEvent-sandl-nonprod.json:20]: I just realised: we might need clarification on what state qualifies this event, and what state the case should be in afterwards (if any specifically)

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]

File: benefit/data/sheets/CaseEvent/CaseEvent-sandl-nonprod.json (hunk 4 above)
Comment (amittal89) [reply benefit/data/sheets/CaseEvent/CaseEvent-sandl-nonprod.json:20]: I checked this with Makeda, the case could be any state to be able to record the outcome.

File: benefit/VERSION.yaml [hunk 5]
```diff
@@ -1 +1 @@
-TAG: 5.15.0
+TAG: S_10167-RC-4
```
Comment (dattipoeHMCTS) [inline benefit/VERSION.yaml:1]: Will need to be set to the appropriate version and in CaseType

File: benefit/data/sheets/CaseField/CaseField-sandl-nonprod.json [hunk 6]
```diff
@@ -3,5 +3,8 @@
   {"LiveFrom": "08/03/2022", "CaseTypeID": "Benefit", "ID": "activeHearingVersionNumber", "Label": "Active Hearing Version Number", "FieldType": "Number", "SecurityClassification": "PUBLIC"},
   {"LiveFrom": "01/06/2022", "CaseTypeID": "Benefit", "ID": "overrideFields", "Label": "Overrides Listing Values", "FieldType": "overrideFields", "SecurityClassification": "PUBLIC"},
   {"LiveFrom": "01/06/2022", "CaseTypeID": "Benefit", "ID": "defaultOverrideFields", "Label": "Default Listing Values", "FieldType": "overrideFields", "SecurityClassification": "PUBLIC"},
-  {"LiveFrom": "06/07/2022", "CaseTypeID": "Benefit", "ID": "amendReasons", "Label": "Amend Reason", "FieldType": "MultiSelectList", "FieldTypeParameter": "FL_amendReason", "SecurityClassification": "PUBLIC"}
+  {"LiveFrom": "06/07/2022", "CaseTypeID": "Benefit", "ID": "amendReasons", "Label": "Amend Reason", "FieldType": "MultiSelectList", "FieldTypeParameter": "FL_amendReason", "SecurityClassification": "PUBLIC"},
+  {"LiveFrom": "04/07/2022", "CaseTypeID": "Benefit", "ID": "caseOutcome", "Label": "Case Outcome", "FieldType": "FixedList", "FieldTypeParameter": "FL_caseOutcome", "SecurityClassification": "PUBLIC"},
```
Comment (dattipoeHMCTS) [inline benefit/data/sheets/CaseField/CaseField-sandl-nonprod.json:7]: I just realised that these two need to be in the prod sheet

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]

File: benefit/data/sheets/CaseField/CaseField-sandl-nonprod.json (hunk 6 above)
Comment (mcareyma) [reply benefit/data/sheets/CaseField/CaseField-sandl-nonprod.json:7]: So 6 and 7 need to be in another json file perhaps CaseEventsToFields.json?

File: benefit/data/sheets/FixedLists/FixedList-sandl-nonprod.json [hunk 7]
```diff
@@ -2,5 +2,15 @@
   {"LiveFrom": "06/07/2022", "ID": "FL_amendReason", "ListElementCode": "partyreq", "ListElement": "Party requested change"},
   {"LiveFrom": "06/07/2022", "ID": "FL_amendReason", "ListElementCode": "judgereq", "ListElement": "Judge requested change"},
   {"LiveFrom": "06/07/2022", "ID": "FL_amendReason", "ListElementCode": "adminreq", "ListElement": "Admin requested change"},
-  {"LiveFrom": "06/07/2022", "ID": "FL_amendReason", "ListElementCode": "adminerr", "ListElement": "Admin error"}
-]
+  {"LiveFrom": "06/07/2022", "ID": "FL_amendReason", "ListElementCode": "adminerr", "ListElement": "Admin error"},
```
[the remark this answers is not in the document]
Comment (mcareyma) [reply benefit/data/sheets/FixedLists/FixedList-sandl-nonprod.json:5]: Sorry, I don't quite follow. Do you mean lines 5 to 15 need to be elsewhere?

File: benefit/VERSION.yaml (hunk 5 above)
Comment (mcareyma) [reply benefit/VERSION.yaml]: I think I have done this but, you may want a larger change in value.

File: benefit/data/sheets/FixedLists/FixedList-sandl-nonprod.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (mcareyma) [reply benefit/data/sheets/FixedLists/FixedList-sandl-nonprod.json:5]: Again in the prod equivalent

[the remark this answers is not in the document]
Comment (dattipoeHMCTS) [reply benefit/data/sheets/FixedLists/FixedList-sandl-nonprod.json:5]: And here the caseOutcomes should be in FixedLists

File: benefit/data/sheets/CaseField/CaseField-sandl-nonprod.json (hunk 6 above)
Comment (mcareyma) [reply benefit/data/sheets/CaseField/CaseField-sandl-nonprod.json]: I have moved CaseOutcome.

File: benefit/data/sheets/FixedLists/FixedList-sandl-nonprod.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (mcareyma) [reply benefit/data/sheets/FixedLists/FixedList-sandl-nonprod.json]: I have moved the caseOutcomes.

File: benefit/data/sheets/CaseEventToFields/CaseEventToFields-sandl-nonprod.json [hunk 8]
```diff
@@ -23,5 +23,47 @@
...
+    "CaseEventID": "caseUpdated",
+    "CaseFieldID": "caseOutcome",
+    "CaseTypeID": "Benefit",
+    "LiveFrom": "05/07/2022",
+    "DisplayContext": "OPTIONAL",
+    "PageDisplayOrder": 1,
+    "PageFieldDisplayOrder": 14,
+    "PageID": "1.0"
+  },
+  {
```
Comment (johnoneill46) [inline benefit/data/sheets/CaseEventToFields/CaseEventToFields-sandl-nonprod.json:59]: I know its not necessary but you've left out "LiveFrom" here if you want to be consistent

Comment (mcareyma) [reply benefit/data/sheets/CaseEventToFields/CaseEventToFields-sandl-nonprod.json:59]: They were there just not at the beginning of the block. I have moved them to the beginning.

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]
Could you remove the audit file if it has no changes please?

File: benefit/data/sheets/CaseType/CaseType.json [hunk 9]
```diff
@@ -1 +1,3 @@
-[{"LiveFrom": "01/01/2018", "ID": "Benefit", "Name": "SSCS Case v5.15.0_${CCD_DEF_E}", "Description": "SSCS Benefit Cases", "JurisdictionID": "SSCS", "PrintableDocumentsUrl": "http://ccd-data-store-api-prod.service.core-compute-prod.internal/callback/jurisdictions/SSCS/case-types/Benefit/documents", "RetriesTimeoutURLPrintEvent": "5,10,15", "SecurityClassification": "PUBLIC"}]
+[{"LiveFrom": "01/01/2018", "ID": "Benefit", "Name": "5.15.1_${CCD_DEF_E}", "Description": "SSCS Benefit Cases", "JurisdictionID": "SSCS", "PrintableDocumentsUrl": "http://ccd-data-store-api-prod.service.core-compute-prod.internal/callback/jurisdictions/SSCS/case-types/Benefit/documents", "RetriesTimeoutURLPrintEvent": "5,10,15", "SecurityClassification": "PUBLIC"}]
```
[the remark this answers is not in the document]
Comment (mcareyma) [reply benefit/data/sheets/CaseType/CaseType.json]: Changed it back

File: benefit/data/sheets/CaseField/CaseField-sandl-nonprod.json [hunk 10]
```diff
@@ -3,5 +3,7 @@
   {"LiveFrom": "08/03/2022", "CaseTypeID": "Benefit", "ID": "activeHearingVersionNumber", "Label": "Active Hearing Version Number", "FieldType": "Number", "SecurityClassification": "PUBLIC"},
   {"LiveFrom": "01/06/2022", "CaseTypeID": "Benefit", "ID": "overrideFields", "Label": "Overrides Listing Values", "FieldType": "overrideFields", "SecurityClassification": "PUBLIC"},
   {"LiveFrom": "01/06/2022", "CaseTypeID": "Benefit", "ID": "defaultOverrideFields", "Label": "Default Listing Values", "FieldType": "overrideFields", "SecurityClassification": "PUBLIC"},
-  {"LiveFrom": "06/07/2022", "CaseTypeID": "Benefit", "ID": "amendReasons", "Label": "Amend Reason", "FieldType": "MultiSelectList", "FieldTypeParameter": "FL_amendReason", "SecurityClassification": "PUBLIC"}
+  {"LiveFrom": "06/07/2022", "CaseTypeID": "Benefit", "ID": "amendReasons", "Label": "Amend Reason", "FieldType": "MultiSelectList", "FieldTypeParameter": "FL_amendReason", "SecurityClassification": "PUBLIC"},
+  {"LiveFrom": "04/07/2022", "CaseTypeID": "Benefit", "ID": "didPoAttend", "Label": "Did PO Attend?", "FieldType": "YesOrNo", "SecurityClassification": "PUBLIC"}
```
[the remark this answers is not in the document]
Comment (mcareyma) [reply benefit/data/sheets/CaseField/CaseField-sandl-nonprod.json:7]: The whole green section?

[the remark this answers is not in the document]
Comment (mcareyma) [reply benefit/data/sheets/CaseField/CaseField-sandl-nonprod.json:7]: Only two lines so I have done it.

[the remark this answers is not in the document]
Comment (mcareyma) [reply benefit/data/sheets/CaseField/CaseField-sandl-nonprod.json:7]: Just the didPAttend line

Comment (dattipoeHMCTS) [inline benefit/data/sheets/CaseField/CaseField-sandl-nonprod.json:7]: didPoAttend is still here

File: benefit/data/sheets/CaseType/CaseType.json [hunk 11]
```diff
@@ -1 +1,3 @@
-[{"LiveFrom": "01/01/2018", "ID": "Benefit", "Name": "SSCS Case v5.15.0_${CCD_DEF_E}", "Description": "SSCS Benefit Cases", "JurisdictionID": "SSCS", "PrintableDocumentsUrl": "http://ccd-data-store-api-prod.service.core-compute-prod.internal/callback/jurisdictions/SSCS/case-types/Benefit/documents", "RetriesTimeoutURLPrintEvent": "5,10,15", "SecurityClassification": "PUBLIC"}]
+[{"LiveFrom": "01/01/2018", "ID": "Benefit", "Name": "SSCS Case v5.15.1_${CCD_DEF_E}", "Description": "SSCS Benefit Cases", "JurisdictionID": "SSCS", "PrintableDocumentsUrl": "http://ccd-data-store-api-prod.service.core-compute-prod.internal/callback/jurisdictions/SSCS/case-types/Benefit/documents", "RetriesTimeoutURLPrintEvent": "5,10,15", "SecurityClassification": "PUBLIC"}]
+
```
Comment (dattipoeHMCTS) [inline benefit/data/sheets/CaseType/CaseType.json:2]: Can you please remove the extra lines

File: benefit/data/sheets/CaseTypeTab/CaseTabType-sandl-nonprod.json [hunk 12]
```diff
@@ -14,5 +14,7 @@
...
     {"LiveFrom": "", "CaseTypeID": "Benefit", "TabID": "listing_requirements_superuser", "TabLabel": "Listing Requirements", "TabDisplayOrder": 24, "CaseFieldID": "overrideFields", "TabFieldDisplayOrder": 11},
-    {"LiveFrom": "", "CaseTypeID": "Benefit", "TabID": "listing_requirements_superuser", "TabLabel": "Listing Requirements", "TabDisplayOrder": 24, "CaseFieldID": "amendReasons", "TabFieldDisplayOrder": 12, "FieldShowCondition": "[STATE]=\"readyToList\""}
-]
+    {"LiveFrom": "", "CaseTypeID": "Benefit", "TabID": "listing_requirements_superuser", "TabLabel": "Listing Requirements", "TabDisplayOrder": 24, "CaseFieldID": "amendReasons", "TabFieldDisplayOrder": 12, "FieldShowCondition": "[STATE]=\"readyToList\""},
+    {"LiveFrom": "", "CaseTypeID": "Benefit","TabID": "outcome_superuser", "TabLabel": "Outcome", "TabDisplayOrder": 30, "CaseFieldID": "caseOutcome", "TabFieldDisplayOrder": 1},
+    {"LiveFrom": "", "CaseTypeID": "Benefit", "TabID": "outcome_superuser", "TabLabel": "Outcome", "TabDisplayOrder": 30, "CaseFieldID": "didPoAttend", "TabFieldDisplayOrder": 2}
```
Comment (dattipoeHMCTS) [inline benefit/data/sheets/CaseTypeTab/CaseTabType-sandl-nonprod.json:19]: Has something gone missing from here? Perhaps another story but we've had bugs/changes raised where only super users could see/action certain things. Do we need more outcome entries for different roles?

**Review by dattipoeHMCTS — CHANGES_REQUESTED** [review]

File: benefit/data/sheets/CaseField/CaseField-sandl-nonprod.json (hunk 10 above)
Comment (mcareyma) [reply benefit/data/sheets/CaseField/CaseField-sandl-nonprod.json:7]: You want that one gone?, I thought it was the other

File: benefit/data/sheets/CaseTypeTab/CaseTabType-sandl-nonprod.json (hunk 12 above)
Comment (mcareyma) [reply benefit/data/sheets/CaseTypeTab/CaseTabType-sandl-nonprod.json:19]: I know that I have not been in this file, But do you want outcomes for judge, clerk and registrar in the same fashion as the super user?

**Review by dattipoeHMCTS — DISMISSED** [review]
Couple more ccd changes are in flight so please make sure your version bump is up to date before merging

**Comment by mcareyma** [comment]
> Couple more ccd changes are in flight so please make sure your version bump is up to date before merging

At present master is on 5.15.0 and this branch is 5.15.1

**Review by ss-evoco — DISMISSED** [review]

**Review by dattipoeHMCTS — DISMISSED** [review]

File: benefit/data/sheets/AuthorisationCaseEvent/AuthorisationCaseEvent-sandl-nonprod.json (hunk 1 above)
Comment (TeaWithLucas) [reply benefit/data/sheets/AuthorisationCaseEvent/AuthorisationCaseEvent-sandl-nonprod.json:7]: Does this need doing?

File: benefit/data/sheets/CaseField/CaseField.json [hunk 13]
```diff
@@ -427,5 +427,7 @@
...
   {"LiveFrom": "08/03/2022", "CaseTypeID": "Benefit", "ID": "hearingState", "Label": "Hearing State", "FieldType": "FixedList", "FieldTypeParameter": "FL_hearingState", "SecurityClassification": "PUBLIC"},
   {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "ID": "workBasketHearingDate", "Label": "Hearing Date", "FieldType": "Date", "SecurityClassification": "PUBLIC"},
-  {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "ID": "workBasketHearingEpimsId", "Label": "Hearing Venue", "FieldType": "FixedList", "FieldTypeParameter": "FL_hearingEpimsId", "SecurityClassification": "PUBLIC"}
+  {"LiveFrom": "19/05/2022", "CaseTypeID": "Benefit", "ID": "workBasketHearingEpimsId", "Label": "Hearing Venue", "FieldType": "FixedList", "FieldTypeParameter": "FL_hearingEpimsId", "SecurityClassification": "PUBLIC"},
+  {"LiveFrom": "04/07/2022", "CaseTypeID": "Benefit", "ID": "caseOutcome", "Label": "Case Outcome", "FieldType": "FixedList", "FieldTypeParameter": "FL_caseOutcome", "SecurityClassification": "PUBLIC"},
+  {"LiveFrom": "04/07/2022", "CaseTypeID": "Benefit", "ID": "didPoAttend", "Label": "Did PO Attend?", "FieldType": "YesOrNo", "SecurityClassification": "PUBLIC"}
```
Comment (TeaWithLucas) [inline benefit/data/sheets/CaseField/CaseField.json:432]: Should these be behind a feature flag?

File: benefit/data/sheets/FixedLists/FixedLists.json [hunk 14]
```diff
@@ -900,5 +900,15 @@
...
+  {"LiveFrom": "04/07/2022", "ID": "FL_caseOutcome", "ListElementCode": "17", "ListElement": "Revised, Assessment increased, for shortened period"},
+  {"LiveFrom": "04/07/2022", "ID": "FL_caseOutcome", "ListElementCode": "21", "ListElement": "Reference, Revised in favour of Appellant"},
+  {"LiveFrom": "04/07/2022", "ID": "FL_caseOutcome", "ListElementCode": "22", "ListElement": "Reference, Revised against Appellant"},
+  {"LiveFrom": "04/07/2022", "ID": "FL_caseOutcome", "ListElementCode": "9", "ListElement": "Decision Upheld"},
+  {"LiveFrom": "04/07/2022", "ID": "FL_caseOutcome", "ListElementCode": "8", "ListElement": "Decision Revised Against Appellant"},
+  {"LiveFrom": "04/07/2022", "ID": "FL_caseOutcome", "ListElementCode": "63", "ListElement": "Decision Reserved"},
+  {"LiveFrom": "04/07/2022", "ID": "FL_caseOutcome", "ListElementCode": "10", "ListElement": "Decision in favour of Appellant"},
+  {"LiveFrom": "04/07/2022", "ID": "FL_caseOutcome", "ListElementCode": "39", "ListElement": "% Disablement Increased <80%, No Benefit award"}
```
Comment (TeaWithLucas) [inline benefit/data/sheets/FixedLists/FixedLists.json:913]: Should these be on non-prod behind a feature flag?

File: benefit/data/sheets/CaseField/CaseField.json (hunk 13 above)
Comment (mcareyma) [reply benefit/data/sheets/CaseField/CaseField.json:432]: Again I was just editing on Del's instructions.

**Comment by mcareyma** [comment]
> Could you remove the audit file if it has no changes please?

What is the audit file?

**Review by dattipoeHMCTS — DISMISSED** [review]

**Review by TeaWithLucas — DISMISSED** [review]

**Review by TeaWithLucas — APPROVED** [review]

**Review by damien-cooke — APPROVED** [review]

**Pull request merged**